File: Android/app/src/main/java/com/rrrfffrrr/unity/kakaotalk/model/TalkProfile.kt
package com.rrrfffrrr.unity.kakaotalk.model

import com.google.gson.annotations.SerializedName
import com.kakao.sdk.talk.model.TalkProfile

data class TalkProfile(
    @SerializedName("nickname") val nickname: String,
    @SerializedName("profile_image_url") val profileImageUrl: String,
    @SerializedName("thumbnail_url") val thumbnailUrl: String,
    @SerializedName("country_iso") val countryISO: String
) {
    constructor(origin: TalkProfile) : this(
        origin.nickname,
        origin.profileImageUrl,
        origin.thumbnailUrl,
        origin.countryISO
    )
}

File: Unity/UnityKakaotalkAPI/Scripts/Models/TalkProfile.cs
using System;
using Newtonsoft.Json;

namespace Kakaotalk.Model {
    [Serializable]
    public class TalkProfile {
        [JsonProperty("nickName")]
        public string nickname;
        [JsonProperty("profileImageURL")]
        public string profile_image_url;
        [JsonProperty("thumbnailURL")]
        public string thumbnail_url;
        [JsonProperty("countryISO")]
        public string country_iso;
    }
}